fineract-investor/src/main/java/org/apache/fineract/investor/cob/loan/LoanAccountOwnerTransferBusinessStep.java

```diff
@@ -0,0 +1,180 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.investor.cob.loan;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.cob.loan.LoanCOBBusinessStep;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.investor.data.ExternalTransferStatus;
+import org.apache.fineract.investor.data.ExternalTransferSubStatus;
+import org.apache.fineract.investor.domain.ExternalAssetOwnerTransfer;
+import org.apache.fineract.investor.service.ExternalAssetOwnersReadService;
+import org.apache.fineract.investor.service.ExternalAssetOwnersWriteService;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class LoanAccountOwnerTransferBusinessStep implements LoanCOBBusinessStep {
+
+    private final ExternalAssetOwnersReadService externalAssetOwnersReadService;
+    private final ExternalAssetOwnersWriteService externalAssetOwnersWriteService;
+
+    @Override
+    public Loan execute(Loan loan) {
+        Long loanId = loan.getId();
+        log.debug("start processing loan ownership transfer business step for loan with Id [{}]", loanId);
+
+        LocalDate settlementDate = DateUtils.getBusinessLocalDate();
+        List<ExternalAssetOwnerTransfer> transferDataList = externalAssetOwnersReadService.retrieveTransfers(loanId, settlementDate,
+                List.of(ExternalTransferStatus.PENDING, ExternalTransferStatus.ACTIVE, ExternalTransferStatus.BUYBACK));
+        int size = transferDataList.size();
+
+        if (size > 2) {
+            throw new IllegalStateException(
```

Comment:
I understand the purpose of this, but don't you think it'd be better to catch this when submitting a transfer?